pkg/controllers/user/helm/controller.go

```diff
@@ -257,6 +261,17 @@ func (l *Lifecycle) Remove(obj *v3.App) (runtime.Object, error) {
 	if err != nil {
 		return obj, err
 	}
+
+	if l.isGatekeeperApp(obj) {
+		logrus.Debugf("Removing GatekeeperApp: %v", obj.Name)
+		//remove the constrainttemplates.
+		err := l.handleConstraintTemplateCleanup(obj)
+		if err != nil {
+			logrus.Errorf("error while deleting gatekeeper constraints for cluster %v: %v", err, l.ClusterName)
```

Comment:
I think if you throw error the framework will print it out. It should be better to use error.Wrapf(). Otherwise the same error will print out twice